File: backend/app/schemas/project.py
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.document import DocumentResponse


class ProjectCreate(BaseModel):
    description: str = Field(
        ...,
        min_length=20,
        max_length=5000,
    )


class WorkflowResponse(BaseModel):
    stage: str
    status: str


class ProjectResponse(BaseModel):
    id: UUID
    title: str
    description: str


class ProjectDetailResponse(BaseModel):
    id: UUID
    title: str
    description: str
    workflow: WorkflowResponse | None = None
    documents: list[DocumentResponse]

File: frontend/src/types/project.ts
export interface Document {

  type: string;

  title: string;

  content: string;

  version: number;

  status: string;

  review_comment: string | null;

}


export interface Project {

  id: string;

  title: string;

  description: string;

}


export interface WorkflowStatus {

  stage: string;

  status: string;

}


export interface ProjectDetail {

  id: string;

  title: string;

  description: string;

  workflow: WorkflowStatus | null;

  documents: Document[];

}


export interface RegenerateRequest {

  review_comment: string;

}